test/hotspot/jtreg/vmTestbase/nsk/monitoring/MemoryPoolMBean/isUsageThresholdExceeded/isexceeded001.java

```diff
@@ -54,17 +54,21 @@ public static int run(String[] argv, PrintStream out) {
 
         for (int i = 0; i < pools.size(); i++) {
             Object pool = pools.get(i);
-            log.display(i + " pool " + monitor.getName(pool) + " of type: " + monitor.getType(pool));
-            if (!monitor.isUsageThresholdSupported(pool)) {
-                log.display("  does not support usage thresholds: skip");
+            // Skip non-heap pools, as they have unpredictable behaviour, or if
+            // usage threshold not supported:
+            if (monitor.getType(pool) != MemoryType.HEAP  || !monitor.isUsageThresholdSupported(pool)) {
                 continue;
             }
+            log.display(i + " pool " + monitor.getName(pool) + " of type: " + monitor.getType(pool));
 
             // Set a threshold that is greater than used value
             MemoryUsage usage = monitor.getUsage(pool);
+            MemoryUsage peakUsage = monitor.getPeakUsage(pool);
             boolean isExceeded = monitor.isUsageThresholdExceeded(pool);
             long used = usage.getUsed();
             long max = usage.getMax();
+            long peakUsed = peakUsage.getUsed();
+            long peakMax = peakUsage.getMax();
```

Comment:
You fetch peakMax here, but you don't display it until after you fetch it again below (for monitor).

Reply:
Actually yes that shows some hesitation where I wanted to take out peakMax,  having stared at the logs it never adds any useful info, we are not stressing the pools such that their max capacity is changing...  I will update, thanks.